pkg/apis/serving/v1alpha2/configmap.go

```diff
@@ -0,0 +1,131 @@
+package v1alpha2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// +k8s:openapi-gen=false
+type ExplainerConfig struct {
+	ContainerImage string `json:"image"`
+
+	DefaultImageVersion  string   `json:"defaultImageVersion"`
+	AllowedImageVersions []string `json:"allowedImageVersions"`
+}
+
+// +k8s:openapi-gen=false
+type ExplainersConfig struct {
+	AlibiExplainer ExplainerConfig `json:"alibi,omitempty"`
+}
+
+// +k8s:openapi-gen=false
+type PredictorConfig struct {
+	ContainerImage string `json:"image"`
+
+	DefaultImageVersion    string   `json:"defaultImageVersion"`
+	DefaultGpuImageVersion string   `json:"defaultGpuImageVersion"`
+	AllowedImageVersions   []string `json:"allowedImageVersions"`
+}
+
+// +k8s:openapi-gen=false
+type PredictorsConfig struct {
+	Tensorflow PredictorConfig `json:"tensorflow,omitempty"`
+	TensorRT   PredictorConfig `json:"tensorrt,omitempty"`
+	Xgboost    PredictorConfig `json:"xgboost,omitempty"`
+	SKlearn    PredictorConfig `json:"sklearn,omitempty"`
+	PyTorch    PredictorConfig `json:"pytorch,omitempty"`
+	ONNX       PredictorConfig `json:"onnx,omitempty"`
+}
+
+// +k8s:openapi-gen=false
+type TransformerConfig struct {
+	ContainerImage string `json:"image"`
+
+	DefaultImageVersion  string   `json:"defaultImageVersion"`
+	AllowedImageVersions []string `json:"allowedImageVersions"`
+}
+
+// +k8s:openapi-gen=false
+type TransformersConfig struct {
+	Feast TransformerConfig `json:"feast,omitempty"`
+}
+
+// +k8s:openapi-gen=false
+type InferenceEndpointsConfigMap struct {
```

Comment:
I think an endpoint is default/canary, and predictor/explainer/etc are called "components". WDYT?

Reply:
Why not just InferenceServiceConfig, TransformerConfig, PredictorConfig, ...